src/ui/public/directives/filter_invert.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash';
+import $ from 'jquery';
+import uiModules from 'ui/modules';
+import invertFilterTemplate from 'ui/directives/partials/filter_invert.html';
+let module = uiModules.get('kibana');
+
+module.directive('filterInvert', function ($compile) {
+  return {
+    restrict: 'A',
+    link: function ($scope, $elem) {
+      /* from modernizer
+       https://raw.githubusercontent.com/Modernizr/Modernizr/master/feature-detects/css/filters.js
+       */
+      function isFilterSupported() {
+        const el = document.createElement('a');
+        el.style.cssText = '-o-filter:blur(2px); -ms-filter:blur(2px); -moz-filter:blur(2px); -webkit-filter:blur(2px); filter:blur(2px);';
+        return !!el.style.length && ((document.documentMode === undefined || document.documentMode > 9));
+      }
+
+      function invert($elem) {
+        $scope.opts = {
+          idHash: (Math.random().toString(36) + '00000000000000000').slice(2, 7),
+          svgWidth: $elem.width(),
+          svgHeight: $elem.height(),
+          svgSource: $elem.attr('src'),
+          className: $elem[0].className
+        };
+
+        $elem.addClass('hide');
+        $elem.after($compile(invertFilterTemplate)($scope));
+      }
+
+      if (!isFilterSupported()) {
+        $elem.css('visibility', 'hidden');
+        $elem.on('load', () => {
+          invert($elem);
+        });
+      } else {
+        $elem.css({
```

Comment:
Is it possible to do partial inversion with the SVG solution, e.g. 50% instead of 100%? If so, it would be cool if this directive took a percent parameter, just like the CSS rule so it's a complete replacement for all our filter inversion needs.